app/models/chargeback.rb

```diff
@@ -21,6 +21,7 @@ def self.build_results_for_report_chargeback(options)
     rate_cols = ChargebackRate.where(:default => true).flat_map do |rate|
       rate.chargeback_rate_details.map(&:metric).select { |metric| perf_cols.include?(metric.to_s) }
     end
+    rate_cols.map! { |x| (x == "v_derived_cpu_total_cores_used") ? "cpu_usage_rate_average" : x }
```

Comment:
@zeari, @lpichler and I looked at this using the DB that you provided and we now understand why you needed to swap out `v_derived_cpu_total_cores_used` for `cpu_usage_rate_average`. We would suggest that you pull that mapping (virtual to real col(s)) out of the code and put it in a constant, instead. This way more mappings can be added there as needed.

Here's the diff of the change we did:

```diff
diff --git a/app/models/chargeback.rb b/app/models/chargeback.rb
index adb0182..8b027a3 100644
--- a/app/models/chargeback.rb
+++ b/app/models/chargeback.rb
@@ -1,4 +1,8 @@
 class Chargeback < ActsAsArModel
+  VIRTUAL_COL_USES = {
+      "v_derived_cpu_total_cores_used" => "cpu_usage_rate_average"
+  }
+
   def self.build_results_for_report_chargeback(options)
     _log.info("Calculating chargeback costs...")
 
@@ -21,7 +25,7 @@ class Chargeback < ActsAsArModel
     rate_cols = ChargebackRate.where(:default => true).flat_map do |rate|
       rate.chargeback_rate_details.map(&:metric).select { |metric| perf_cols.include?(metric.to_s) }
     end
-    rate_cols.map! { |x| (x == "v_derived_cpu_total_cores_used") ? "cpu_usage_rate_average" : x }
+    rate_cols.map! { |x| VIRTUAL_COL_USES.include?(x) ? VIRTUAL_COL_USES[x] : x }.flatten!
     base_rollup = base_rollup.select(*rate_cols)
 
     timerange = get_report_time_range(options, interval, tz)
```